codalab/apps/authenz/models.py

```diff
@@ -32,4 +41,57 @@ class ClUser(auth_models.AbstractUser):
     rabbitmq_username = models.CharField(max_length=36, null=True, blank=True)
     rabbitmq_password = models.CharField(max_length=36, null=True, blank=True)
 
+    def save(self, *args, **kwargs):
```

Comment:
We should check to make sure we have all the mailchimp settings set before we try and make mailchimp requests.
We also need to catch exceptions from `requests`.  I believe `requests.exceptions.RequestException` is the catch-all exception. 
We should probably do something like the `chahub_needs_retry` logic so we make sure updates actually get through to mailchimp, _especially_ in the case of an unsubscribe.

Reply:
So thinking about this a little more completely, I think we should move this business logic to the `NewsletterUser` model. That model should have like a `subscribe` and `unsubscribe` method, which will calculate the hash itself (maybe even store the hash as a field on that model? or maybe a `@property`?) as well as handle request exceptions and all that. This will clean up the `User` model `save` logic quite a lot. Would probably reduce a lot of DRY violation throughout the code as well.

Reply:
Fixed in latest push